sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala

```diff
@@ -86,7 +86,7 @@ class CheckAnalysis {
             cleaned.foreach(checkValidAggregateExpression)
 
           case o if o.children.nonEmpty &&
-            !o.references.filter(_.name != "grouping__id").subsetOf(o.inputSet) =>
```

Comment:
@chenghao-intel I think the resolver had been used in `case u @ UnresolvedAttribute(name) if resolver(name, VirtualColumn.groupingIdName)`, so the string is already in lower case.